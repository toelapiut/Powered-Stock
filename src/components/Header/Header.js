import React from 'react';
import PropTypes from 'prop-types';
import DateInput from '../../components/DateInput';
import styles from './header.module.css';



export const Header = ({name, date, isOpen, onChangeDates, ticker, start, end, onOpenCalendar}) => {
  return (
    <div className={styles.container}>
      <div>
        <h1 className={styles.ticker}>{ticker}</h1>
        <p className={styles.name}>{name}</p>
      </div>
      <DateInput
        date={date}
        isOpen={isOpen}
        onChangeDates={onChangeDates}
        start={start}
        end={end}
        onOpenCalendar={onOpenCalendar}
      />
    </div>
  );
};

Header.propTypes = {
  date: PropTypes.shape({
    start: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    end: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ])
  }),
  name:PropTypes.string.isRequired,
  ticker:PropTypes.string.isRequired,
  start: PropTypes.string,
  end: PropTypes.string,
  isOpen: PropTypes.bool.isRequired,
  onChangeDates: PropTypes.func.isRequired,
  onOpenCalendar: PropTypes.func.isRequired,
};

