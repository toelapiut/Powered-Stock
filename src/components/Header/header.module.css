.container {
    @apply bg-koamaru pl-8 pr-4 py-4 rounded-lg mb-5 mt-6 flex flex-row items-center justify-between
}
.details{

}

.ticker{
    @apply text-3xl font-black text-white leading-10
}

.name{
    @apply text-sm font-bold text-liberty
}