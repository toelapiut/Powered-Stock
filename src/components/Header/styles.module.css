.container {
    @apply pt-10 px-10 mb-5
}

.title {
    @apply text-3xl font-black text-white
}


.header {
    @apply text-koamaru font-bold text-base;
}