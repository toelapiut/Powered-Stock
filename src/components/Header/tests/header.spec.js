import React from 'react';
import Header from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const setUp = (props = {}) => {
  return shallow(<Header {...props}/>);
};

describe('Header Component', () => {
  let component;
  beforeAll(() => {
    let props = {
      title: 'Market',
    };
    component = setUp(props);
  });

  test('should render snapshot correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});