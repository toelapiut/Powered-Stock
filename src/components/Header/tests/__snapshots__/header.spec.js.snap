// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render snapshot correctly 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    Market
  </h1>
  <p
    className="header"
  >
    Powered by People
  </p>
</div>
`;
